pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -195,8 +206,7 @@ stdenv.mkDerivation (finalAttrs: {
     ++ lib.optional capstoneSupport "--enable-capstone"
     ++ lib.optional (!pluginsSupport) "--disable-plugins"
     ++ lib.optional (!enableBlobs) "--disable-install-blobs"
-    ++ lib.optional userOnly "--disable-system"
-    ++ lib.optional stdenv.hostPlatform.isStatic "--static";
```

Comment:
Why?

Reply:
otherwise meson cannot find libaio

Reply:
Why not?

I suspect that this might be the reason for the pam thing not being propagated — if Meson isn't told to do a static build, it won't ask pkg-config for private dependencies.

Reply:
well I don't know. Here is an excerpt of meson.log:
```
Running compile:
Working directory:  /build/qemu-9.1.0/build/meson-private/tmpyireszaj
Code:

        #ifdef __has_include
         #if !__has_include("libaio.h")
          #error "Header 'libaio.h' could not be found"
         #endif
        #else
         #include <libaio.h>
        #endif
-----------
Command line: `x86_64-unknown-linux-musl-gcc -m64 /build/qemu-9.1.0/build/meson-private/tmpyireszaj/testfile.c -E -P -D_FILE_OFFSET_BITS=64 -P -O0 -std=gnu11` -> 0
Has header "libaio.h" : YES 

../meson.build:1031:14: ERROR: C prefer_static library 'aio' not found
```
corresponding to https://github.com/qemu/qemu/blob/master/meson.build#L1045

It might be that cc.find_library is broken in meson when prefer_static is true.